### ISSUE TITLE
Lack of Move Option When Window is Stuck Off Screen

### ISSUE THREAD

**Issue by Psalm27-4** [mattermost/desktop#1736]
Hi,

This is for the Windows 10 Mattermost Desktop App, Version 5.36.0. I am observing that Mattermost can become "stuck" off screen when changing setups with multiple monitors. A couple days a week I transition into the office from home, this involves bringing my dock and laptop in. I have two monitors plugged into my dock in both settings, which are both different manufacturer than those I have at home. In both setups, the laptop is on the far left of both monitors, with the display settings set to mimic that setup.

When transitioning to and from either place, the first time I start Mattermost after power up it automatically goes off the side of the far right monitor. Stopping and starting Mattermost also results in the window being created in the same unreachable spot. I cannot select click on it when it is off screen, clicking on its taskbar only raises and lowers the window in its unreachable spot. Closing the app via the Task Manager doesn't fix it. Only rebooting a few times.

In most applications, a user can shift-right-click on the app in the taskbar, and click on move in the popup context menu, followed by using the arrow keys to move the application back onto the visible screen. Unfortunately Mattermost lacks this option.

Can this be added/fixed, or is there a way to clear the last saved window position in the registry, so Mattermost will spawn the window on a visible part of the screen?

**Comment by svelle** [comment]
Hey thanks for raising this, I've experienced this myself a couple of times before so would really appreciate some improvements in this regard.

However what also helped me in the past was Win + Arrow keys moving the window back to one of the active working screens, hope this helps you too!

**Comment by Willyfrog** [comment]
@svelle suggestion is spot on! thanks!

But if everything else fails, you can always go to the config folder of your mattermost desktop installation and delete the file bounds-info.json which is the one responsible for saving window state between restarts.

**Comment by Psalm27-4** [comment]
Unfortunately that doesn't work for me. However, the Windows + Down key and Windows + Up Key do show the minimize and raise animations happening. The Windows + Left or Right Arrow keys don't move it, until something happens after a few reboots to fix the issue.

**Comment by devinbinnie** [comment]
@Psalm27-4 Thanks for reporting this, I have also had this happen to me. I think what's causing it has to do with the fact that we already do save the last window position in a file called `bounds-info.json` in your AppData directory. Unfortunately if you change monitor configurations after the app has been closed and before you re-open it, it will still use that data which may not be valid anymore.

I think what needs to happen is that we need to make a fix to make sure that the part of the screen that the window tries to render on is still valid before we use it. I've created a ticket to track this: https://mattermost.atlassian.net/browse/MM-38607
